### PR TITLE
internal: Add a manual section for Flatpak

### LINKED ISSUE #14042: rust analyzer failed to load workspace
Error:
```ini
rust-analyzer failed to load workspace: Failed to read Cargo metadata from Cargo.toml file /mnt/SSD/Code/Porn Folder/test_cli_animation/Cargo.toml,
Some(Version { major: 1, minor: 65, patch: 0 }): Failed to run `"/home/electricthunder/.cargo/bin/cargo" "metadata" "--format-version" "1" "--manifest-path" "/mnt/SSD/Code/Porn Folder/test_cli_animation/Cargo.toml" "--filter-platform" "x86_64-unknown-linux-gnu"`: `cargo metadata` exited with an error:
error: could not execute process `rustc -vV` (never executed)

Caused by:
  No such file or directory (os error 2)
```
This error does not seem to want to go away. I tried to add
```json
    "rust-analyzer.server.extraEnv": {
        "CARGO": "~/.cargo/bin/cargo",
    },
```
to the settings.json, but now I get:
```ini
rust-analyzer failed to load workspace: "~/.cargo/bin/cargo" "--version" failed: No such file or directory (os error 2)
```

I am using VSCodium from flatpak and I thought I didn't give it the correct permissions to access cargo but running
```sh
bash-5.1$ cargo -V
cargo 1.65.0 (4bc8f24d3 2022-10-20)
```
in the VSCodium terminal outputs the version, not an error, which means VSCodium has access to the binary.
So I have no idea what to do now. Any pointers?
Cheers!

### PR DESCRIPTION
**Pull request by lnicola**
Fixes #14042

### CODE REVIEW

File: docs/user/manual.adoc [hunk 1]
```diff
@@ -232,6 +232,26 @@ The `rust-analyzer` binary can be installed via https://brew.sh/[Homebrew].
 $ brew install rust-analyzer
 ----
 
+=== VS Code or VSCodium in Flatpak
+
+We do not recommend using `rust-analyzer` with a Flatpak version of Code.
+Flatpak applications are sandboxed, which has some unfortunate downsides.
+While the sandbox can be disabled for some directories, `/usr/bin` will always be mounted under `/run/host/usr/bin`.
+This prevents applications from running the system's C compiler, or a system-wide installation of Rust.
```
Comment (bjorn3) [inline docs/user/manual.adoc:240]: The `org.freedesktop.Sdk` runtime (which both vscode and vscodium has a dependency on) contains a C compiler. There is also the `org.freedesktop.Sdk.Extension.rust-stable` sdk extension containing rustc, but that one you need to install manually. I personally use vscode in flatpak with a rustup installed rustc without issues every day. The only time I had trouble was when I used rust-bindgen as libclang is missing, but I just found that `org.freedesktop.Sdk.Extension.llvm15` exists which I think has libclang.

Comment (DanielProg39) [reply docs/user/manual.adoc:240]: If everything is great and clear, then what is the issue behind #14042 and #13926?

**Comment by DanielProg39** [comment]
Idk how well does this works in terms of resolving my and @ionutrogojan's issues, but seems legit. Will test as soon as will have time. Thanks for mentioning and letting me know 😉

**Review by DanielProg39 — APPROVED** [review]

File: docs/user/manual.adoc (hunk 1 above)
Comment (bjorn3) [reply docs/user/manual.adoc:240]: I'm using a rustup installed rustc and I've configured the `PATH` env var to add `/home/bjorn/.cargo/bin:` before the default value. Other than that I don't have any flatpak overrides. I also couldn't find any relevant differences between the vscode and vscodium flatpak.

File: docs/user/manual.adoc [hunk 2]
```diff
@@ -232,6 +232,26 @@ The `rust-analyzer` binary can be installed via https://brew.sh/[Homebrew].
...
+
+If you insist on using Flatpak, the following steps might help:
+
+ - both Rust and `rustup` need to be installed using https://rustup.rs. Distro packages _will not_ work.
+ - before installing `rust-analyzer`, you need to launch Code, open a terminal, run `echo $PATH`
```
Comment (ionutrogojan) [inline docs/user/manual.adoc:245]: This works if you already have the extension installed too, you just need to restart VSCode/VSCodium so the Flatseal changes take effect

**Comment by ionutrogojan** [comment]
I tried [the fix](https://github.com/rust-lang/rust-analyzer/issues/14042#issuecomment-1407583506), it works perfectly and it's very easy to setup too. The only setting missing from what I had before was the `PATH` variable which makes it work.
**+1** from me

**Review by ionutrogojan — APPROVED** [review]

File: docs/user/manual.adoc (hunk 1 above)
Comment (lnicola) [reply docs/user/manual.adoc]: @DanielProg39 in #14042, @ionutrogojan was using Rust installed with `rustup`, but without adding it to Codium' `PATH`.

I assume you were in the same situation in #13926.

The "Flatpak-native" solution would be to uninstall `rustup` and Rust, and install `org.freedesktop.Sdk.Extension.rust-stable` from Flatpak.

**Comment by lnicola** [comment]
Switched to a more precise and less scary wording, recommending the Flatpak SDK instead of changing `PATH`. Thanks @bjorn3!

**Pull request merged**